File: array-problem/100001_find-the-sum-of-all-elements-in-an-array_array-problem_beginner.c
/* DSA_PLATFORM_HEADER_V1
 * Problem: Array Problem
 * Unique ID: 100001
 * Difficulty: beginner
 * Date: 2026-07-25
 * Source: DSA Practice Platform
 */

#include <stdio.h>  // For printf, scanf

int main() {
    // Read the number of elements N (0 ≤ N ≤ 10^5)
    int N;
    // Use scanf to read an integer from standard input
    scanf("%d", &N);

    // Use a 64-bit integer to store the sum (max possible sum 10^14 fits in long long)
    // In C, long long is a 64-bit signed integer type, same as C++'s long long
    long long sum = 0;

    // If N > 0, read the N integers one by one
    if (N > 0) {
        // Loop exactly N times to read each element
        for (int i = 0; i < N; ++i) {
            // Temporary variable to hold the current array element
            int x;
            // Read the next integer from standard input
            scanf("%d", &x);

            // Add the current element to the running sum
            sum += x;
        }
    }
    // If N == 0, the second line is absent; sum remains 0 (correct for empty array)

    // Output the total sum followed by a newline
    // Use %lld format specifier for long long in printf
    printf("%lld\n", sum);

    return 0;  // Indicate successful program termination
}

File: array-problem/100001_find-the-sum-of-all-elements-in-an-array_array-problem_beginner.py
/* DSA_PLATFORM_HEADER_V1
 * Problem: Array Problem
 * Unique ID: 100001
 * Difficulty: beginner
 * Date: 2026-07-25
 * Source: DSA Practice Platform
 */

import sys  # Import sys for reading from standard input

def main():
    # Read the first line which contains the integer N (number of elements)
    # Using sys.stdin.readline() to read one line at a time
    first_line = sys.stdin.readline()
    # If the input is empty (e.g., EOF), treat N as 0
    if not first_line:
        N = 0
    else:
        # Strip whitespace (newline) and convert to integer
        N = int(first_line.strip())

    # Initialize sum as an integer (Python int is arbitrary precision,
    # so no overflow issues even with 10^14)
    total_sum = 0

    # Only read the second line if N > 0
    if N > 0:
        # Read the second line containing N space-separated integers
        second_line = sys.stdin.readline()
        # If the line exists, split it into tokens and parse each token
        if second_line:
            # Split the line by whitespace (default split handles multiple spaces)
            tokens = second_line.split()
            # Ensure we have at least N tokens (in case of extra spaces, safe)
            # Loop over the first N tokens (in case of more tokens, ignore extras)
            for i in range(N):
                # Convert token to integer and add to total sum
                total_sum += int(tokens[i])
        # If the second line is missing (should not happen for valid input),
        # treat as if no numbers (sum stays 0) – but problem guarantees presence.
    # If N == 0, the loop is skipped, total_sum remains 0 (correct for empty array)

    # Print the total sum followed by a newline
    # Using sys.stdout.write for consistency, but print is also fine
    sys.stdout.write(str(total_sum) + "\n")

# Call the main function when the script is executed
if __name__ == "__main__":
    main()